### PR TITLE
Feature/aa 1450 giata accommodations matching

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

File: HappyTravel.Nakijin.Updater/Models/Giata/Matching/ChangedData.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+using System.Collections.Generic;
+using HappyTravel.Nakijin.Data.Models.Enums;
+
+namespace HappyTravel.Nakijin.Updater.Models.Giata.Matching;
+
+public record  ChangedData
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Updater/Models/Giata/Matching/ChangedData.cs:6]: Extra space

File: HappyTravel.Nakijin.Data/Migrations/NakijinContextModelSnapshot.cs [hunk 2]
```diff
@@ -248,18 +251,52 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
 
                     b.Property<string>("Supplier")
-                        .IsRequired()
                         .HasColumnType("text");
 
                     b.Property<int>("Type")
                         .HasColumnType("integer");
 
-                    b.Property<DateTimeOffset>("UpdateTime")
+                    b.Property<DateTimeOffset>("UpdateDate")
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Data/Migrations/NakijinContextModelSnapshot.cs:259]: Just Updated maybe?

File: HappyTravel.Nakijin.Data/Models/Enums/DeactivationReasons.cs [hunk 3]
```diff
@@ -9,11 +9,12 @@ public enum DeactivationReasons
         InvalidCoordinates = 1,
         InvalidName = 2,
         DuplicateInOneSupplier = 4,
-        MatchingWithOther = 8,
+        MergeWithOther = 8,
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Data/Models/Enums/DeactivationReasons.cs:12]: "MergedWithAccommodations" ?

File: HappyTravel.Nakijin.Updater/Infrastructure/GiataDataProviderClient.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        _httpRequestProcessor = httpRequestProcessor;
+        _options = options.Value;
+    }
+
+
+    public Task<Result<List<Accommodation>, ProblemDetails>> GetAccommodations(int skip, int top, DateTimeOffset modifiedDate,
+        CancellationToken cancellationToken)
+    {
+        var url = _options.Endpoint +
+            $"{AccommodationUrl}?skip={skip}&top={top}&modification-date={System.Net.WebUtility.UrlEncode(modifiedDate.ToString())}";
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Updater/Infrastructure/GiataDataProviderClient.cs:27]: AccommodationSectionOfUrl ?

File: HappyTravel.Nakijin.Updater/Infrastructure/HttPRequestProcessor.cs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace HappyTravel.Nakijin.Updater.Infrastructure
+{
+    public class HttpRequestProcessor 
+    {
+        public HttpRequestProcessor(IHttpClientFactory clientFactory, ILogger<HttpRequestProcessor> logger)
+        {
+            _serializer = new JsonSerializer();
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Updater/Infrastructure/HttPRequestProcessor.cs:21]: Can be set on declaration of the variable

File: HappyTravel.Nakijin.Updater/Workers/Giata/Matching/DataStorageForGiataAccommodationMatching.cs [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        await InitializeHtAccommodations(countryCode, cancellationToken);
+        await InitializeUncertainMatches(countryCode, cancellationToken);
+        await InitializeHtAccommodationMappings(countryCode, cancellationToken);
+    }
+
+
+    private async Task InitializeGiataAccommodations(string countryCode, DateTimeOffset lastMatchDate, CancellationToken cancellationToken)
+    {
+        _giataAccommodations = await _context.GiataAccommodations
+            .Where(ac => ac.CountryCode == countryCode && ac.Modified > lastMatchDate)
```
Comment (oleg-n-k) [inline HappyTravel.Nakijin.Updater/Workers/Giata/Matching/DataStorageForGiataAccommodationMatching.cs:41]: && isActive ?

Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Updater/Workers/Giata/Matching/DataStorageForGiataAccommodationMatching.cs:41]: We get from connector only active accommodations, activity change will be in separate place, by provided data from giata about hotels merge.

File: HappyTravel.Nakijin.Data/Migrations/NakijinContextModelSnapshot.cs (hunk 2 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Data/Migrations/NakijinContextModelSnapshot.cs:259]: This is discussed, `Updated` we use for entity updates, here `UpdateDate` is `Updater` work date (data update date).

File: HappyTravel.Nakijin.Data/Models/Enums/DeactivationReasons.cs (hunk 3 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Data/Models/Enums/DeactivationReasons.cs:12]: I think this kind of changes we can do later

**Comment by hasmikharutyunyan** [comment]
Changes will be merged from another PR

**Pull request closed without merging**